packages/melos/lib/src/command_runner/analyze.dart

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2016-present Invertase Limited & Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this library except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import 'dart:io';
+
+import '../commands/runner.dart';
+import '../common/utils.dart';
+import 'base.dart';
+
+class AnalyzeCommand extends MelosCommand {
+  AnalyzeCommand(super.config) {
+    setupPackageFilterParser();
+    argParser.addOption('concurrency', defaultsTo: '1', abbr: 'c');
+    argParser.addFlag(
+      'fatal-infos',
+      negatable: false,
+      help: 'Enables treating info-lever issues as fatal errors,'
+          ' stopping the process if any are encountered.',
+    );
+
+    argParser.addFlag(
+      'fatal-warnings',
+      help: 'Enables or disables treating warnings as fatal errors. '
+          'When enabled, any warning will cause the command to fail.',
+    );
+  }
+
+  @override
+  final String name = 'analyze';
+
+  @override
+  final String description =
+      '''Analyzes all packages in your project for potential issues in a single run. Optionally configure severity levels. Supports all package filtering options.''';
+
+  @override
+  Future<void> run() async {
+    if (argResults!.rest.isNotEmpty) {
+      logger.log('No additional command line arguments are needed');
```

Comment:
This might be a bit confusing if one of the valid arguments are used and then an invalid one is also used, maybe list the invalid ones?

Reply:
I think it's pretty hard to come up with a list of invalid arguments because it can be anything, right? That's why I chose to remove this code. Even if the user tries to pass an extra argument, it simply won't work, and the 'melos analyze' command will run with the 'default' behavior, which means it will analyze every package or accept a valid argument. For example ⬇️ 
<img width="1004" alt="Screenshot 2024-03-10 at 20 50 16" src="https://github.com/invertase/melos/assets/65210800/9e4d0430-ff09-467f-b9ea-5a8c65dc565e">

Also, it's important to mention that this made me wonder what would happen if the user combines two arguments, like `--fatal-infos` and `--fatal-warnings`. So, I introduced the following commit: e66abd61e1b9b2e41cb28b9fd0358989bde6936d

Reply:
Since it has the list of user arguments it could just print the invalid ones for the user, but I think it was better to remove it like you did anyways, then it's more in line with how the other commands look like.